jax/_src/numpy/lax_numpy.py

```diff
@@ -846,15 +846,13 @@ def unravel_index(indices: ArrayLike, shape: Shape) -> tuple[Array, ...]:
   indices_arr = asarray(indices)
   # Note: we do not convert shape to an array, because it may be passed as a
   # tuple of weakly-typed values, and asarray() would strip these weak types.
-  try:
-    shape = list(shape)
-  except TypeError:
-    shape = [shape]
+  shape = list(shape)
```

Comment:
`shape` is annotated `Sequence`, so the`list` call can never fail.

Reply:
This is an unsafe change. While true that it can never fail at type-checking time, runtime types may (and often do) differ from their annotations.

Reply:
I understand your point, but something is very awkward about the code that's written.  The interface says that it accepts `Shape` only.  I can understand that you don't want to break code that's relying on calling the function with things other than `Shape`.

If you don't want to disallow calling the function with non-shapes, then at least I think you should:
* raise a warning to induce callers to call the function with an actual shape, and
* assert on some wider interface to catch problems as early as possible.

The reason to assert is to prevent someone passing some random thing `x` (an array, a tensor, who knows), and then it gets turned into `[x]`, and then causes some weird behavior down the line.  Better to assert that `x` is an `int` or `None` or whatever else you want to accept for the time being?